Contest_API.md

```diff
@@ -689,13 +689,13 @@ Properties of contest objects:
 | :--------------------------- | :-------------- | :----------
 | id                           | ID              | Identifier of the current contest.
 | name                         | string          | Short display name of the contest.
-| formal\_name                 | string          | Full name of the contest.
+| formal\_name                 | string ?        | Full name of the contest.
```

Comment:
Should have a default.

```suggestion
| formal\_name                 | string ?        | Full name of the contest. Defaults to value of `name`.
```

Reply:
I hesitated with this because what we said before is that clients should fall back to using name (i.e. 'the contest doesn't have a formal name and you should use the name for display purposes'), which isn't quite the same as defaulting formal_name to name. To me this risks someone writing 'if (formal_name == null) then formal_name = name' or saying 'the contest has both properties and they're identical'.

Reply:
To me "clients should fall back to using name" means the same as "defaulting formal_name to name". 

If we don't say that formal_name defaults to name, then if formal_name is missing that means that there is no formal_name, and it must be up to the client to choose how to handle that, we shouldn't tell the client what it "should" do.

Reply:
Would "If formal_name does not exist use name for display purposes" be enough? I'm fine telling the client what to do in this case, I just don't like (possibly due to history?) formal_name picking up the value of name.

Reply:
I think given the discussion we had, this should not be nullable: either the server has the formal name and then should specify that in `/access` and it should be non-null, or it doesn't, in which case it shouldn't list it in `/access` and then it must not be present.